3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/AnonIpEnrichment.scala

```diff
@@ -118,23 +167,60 @@ case class AnonIpEnrichment(
    * anonymizeIp("94.15.223.151", Three)
    * => "94.x.x.x"
    *
+   *
+   * For IPv6 either the form defined in RFC 2732
+   *  or the literal IPv6 address format defined in RFC 2373 is accepted
+   *
    * TODO: potentially update this to return
    * a Validation error or a null if the IP
    * address is somehow invalid or incomplete.
    *
    * @param ip The IP address to anonymize
    * @return the anonymized IP address
    */
-  import AnonOctets._
-  def anonymizeIp(ip: String): String =
-    Option(ip)
-      .map(
-        _.split("\\.").zipWithIndex
-          .map {
-            case (q, i) => {
-              if (octets.id >= All.id - i) "x" else q
-            }
-          }
-          .mkString("."))
-      .orNull
+  val IPv4MappedAddressPrefix = "::FFFF:"
+  val MaskChar                = "x"
+
+  def anonymizeIp(ipOrNull: String): String =
+    Option(ipOrNull).map { ip =>
+      Try(InetAddress.getByName(ip))
+        .map {
+          case _: Inet4Address => anonymizeIpV4(ip)
+          case _: Inet6Address => anonymizeIpV6(ip)
+        }
+        .getOrElse(tryAnonymizingInvalidIp(ip))
+    }.orNull
+
+  private def anonymizeIpV4(ip: String): String = {
+    def masked(ipv4: String) = {
+      val splitted = ipv4.split("\\.")
+      splitted
+        .take(AnonIPv4Octets.All.id - ipv4Octets.id)
+        .toList
+        .padTo(splitted.size, MaskChar)
+        .mkString(".")
+    }
+
+    if (ip.startsWith(IPv4MappedAddressPrefix))
+      IPv4MappedAddressPrefix + masked(ip.replace(IPv4MappedAddressPrefix, ""))
+    else masked(ip)
+  }
+
+  private def anonymizeIpV6(ip: String): String =
+    InetAddress
+      .getByName(ip)
```

Comment:
As far as I understand it makes blocking DNS IO call. Can we avoid that?

Reply:
If it's IP, not host name it doesn't do the call. I have commented it after opening PR, https://github.com/snowplow/snowplow/pull/4223#pullrequestreview-302575456

Reply:
@benjben do you know if R118 will still have guava as a dependency?

Reply:
I added Guava explicitly to `Dependencies`, but it of course was already as some transitive dependency. The versions didn't clash also.

Reply:
Just in case - that's a tricky trade-off - we don't want to eventually end up with a guava dependency _just_ for IPs (which we probably won't know). But probably that's still better than potential IO.

Reply:
I can't find a use of guava in SCE for R118. But as long as it's there as a transitive dependency, why not using it.